src/app/modules/ix-table2/components/ix-table-body/cells/ix-cell-actions/icon-action-config.interface.ts

```diff
@@ -4,9 +4,10 @@ import { Role } from 'app/enums/role.enum';
 export interface IconActionConfig<T> {
   iconName: string;
   tooltip?: string;
+  requiredRoles?: Role[];
   onClick: (row: T) => void;
-  dynamicTooltip?: (row: T) => Observable<string>;
   hidden?: (row: T) => Observable<boolean>;
   disabled?: (row: T) => Observable<boolean>;
-  requiredRoles?: Role[];
+  dynamicTooltip?: (row: T) => Observable<string>;
+  dynamicRequiredRoles?: (row: T) => Observable<Role[]>;
```

Comment:
I'm curious about maintainability with two inputs for a single parameter. I see two tooltips already there, but I wouldn't say I like this approach, which keeps growing. Maybe we could check if it is observable and pass it forward, and if it is not, make it observable on the fly to keep a single input. What do you think?

Reply:
let's consider it for the future ticket and leave it as it is here.